prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java

```diff
@@ -0,0 +1,314 @@
+        private ThreadPoolExecutor executor;
+
+        @Override
+        public void run() {
+            executor = new ScheduledThreadPoolExecutor(10);
+            List<Future> futures = new ArrayList<>();
+            try {
+                for (Entity entity : entityBacklogs.keySet()) {
+                    futures.add(executor.submit(new BacklogCalcService(entity, entityBacklogs.get(entity))));
+                }
+                waitForFuturesToComplete(futures);
+            } finally {
+                executor.shutdown();
+            }
+        }
+
+        private void waitForFuturesToComplete(List<Future> futures) {
+            try {
+                for (Future future : futures) {
+                    future.get();
+                }
+            } catch (InterruptedException e) {
+                LOG.error("Interruption while executing tasks " + e);
+            } catch (ExecutionException e) {
+                LOG.error("Error in executing threads " + e);
+            }
+        }
+    }
+
+    /**
+     * Service which calculates backlog for given entity and publish to graphite.
+     */
+    public static class BacklogCalcService implements Runnable {
+
+        private Entity entityObj;
+        private List<MetricInfo> metrics;
+
+        BacklogCalcService(Entity entity, List<MetricInfo> metricInfoList) {
+            this.entityObj = entity;
+            this.metrics = metricInfoList;
+        }
+
+        @Override
+        public void run() {
+            long backlog = 0;
+            MetricInfo metricInfo = null;
+            synchronized (metrics) {
+                long currentTime = System.currentTimeMillis();
+                Iterator iter = metrics.iterator();
+                while (iter.hasNext()) {
+                    try {
+                        metricInfo = (MetricInfo) iter.next();
+                        long time = DATE_FORMAT.get().parse(metricInfo.getNominalTime()).getTime();
+                        backlog += (currentTime - time);
+                    } catch (ParseException e) {
+                        LOG.error("Unable to parse nominal time" + metricInfo.getNominalTime());
+                    }
+                }
+            }
+            org.apache.falcon.entity.v0.process.Process process = (Process) entityObj;
+
+            String metricName = METRIC_PREFIX + METRIC_SEPARATOR + metricInfo.getCluster() + METRIC_SEPARATOR
+                    + process.getPipelines() + METRIC_SEPARATOR + LifeCycle.EXECUTION.name()
+                    + METRIC_SEPARATOR + entityObj.getName() + METRIC_SEPARATOR
+                    + "backlogInMins";
+            backlog = backlog / (60 * 1000); // Converting to minutes
+            metricNotificationService.publish(metricName, backlog);
+        }
+    }
+
+
+    /**
+     * Service runs periodically and removes succeeded instances from backlog list.
+     */
+    public static class BacklogCheckService implements Runnable {
```

Comment:
This will be useful in few cases like when we get success notification and deleted entry from state store but before deleting from memory if falcon was restarted. Then that stale will instance will stuck forever this service will be useful to remove it from memory.